jobs/async-upload/tests/test_models.py

```diff
@@ -0,0 +1,273 @@
+            )
+        
+        error_message = str(exc_info.value)
+        assert "Model Artifact ID must be set when intent is update_artifact" in error_message
+
+    def test_default_intent_is_update_artifact(self):
+        """Test that the default upload intent is update_artifact"""
+        config = ModelConfig(
+            id="test-model-id",
+            version_id="test-version-id",
+            artifact_id="test-artifact-id"
+        )
+        
+        # Default should be update_artifact
+        assert config.upload_intent == UploadIntent.update_artifact
+
+    def test_default_intent_validation_with_valid_artifact_id(self):
+        """Test that default intent (update_artifact) validates correctly with artifact ID"""
+        config = ModelConfig(
+            id="test-model-id",
+            version_id="test-version-id",
+            artifact_id="test-artifact-id"
+        )
+        
+        # Should not raise any validation errors with default intent
+        assert config.upload_intent == UploadIntent.update_artifact
+        assert config.artifact_id == "test-artifact-id"
+
+    def test_default_intent_validation_fails_without_artifact_id(self):
+        """Test that default intent (update_artifact) validation fails without artifact ID"""
+        with pytest.raises(ValidationError) as exc_info:
+            ModelConfig(
+                id="test-model-id",
+                version_id="test-version-id",
+                artifact_id=""  # Empty string
+            )
+        
+        error_message = str(exc_info.value)
+        assert "Model Artifact ID must be set when intent is update_artifact" in error_message
+
+    def test_pydantic_field_validation_prevents_none_values(self):
+        """Test that Pydantic field validation prevents None values for required fields"""
+        # Pydantic should prevent None values due to Field(...) constraints
+        with pytest.raises(ValidationError):
+            ModelConfig(
+                upload_intent=UploadIntent.update_artifact,
+                id="test-model-id",
+                version_id="test-version-id",
+                artifact_id=None  # This should be caught by Pydantic field validation
+            )
+
+    def test_whitespace_only_strings_behavior(self):
+        """Test behavior with whitespace-only strings for different intents"""
+        # create_model: No validation, so whitespace is fine
+        config1 = ModelConfig(
+            upload_intent=UploadIntent.create_model,
+            id="   ",  # Whitespace only - fine for create_model
+            version_id="   ",
+            artifact_id="   "
+        )
+        assert config1.upload_intent == UploadIntent.create_model
+        
+        # create_version: Only validates model ID, whitespace-only string is truthy
+        config2 = ModelConfig(
+            upload_intent=UploadIntent.create_version,
+            id="   ",  # Whitespace only - truthy, so validation passes
```

Comment:
do we really want this behaviour? Shouldn't we sanitize input with strip() for example?